### PR TITLE
Update rubocop config

### PR DESCRIPTION
**Pull request by allenwq**
- Increase max line length, max AbcSize and max MethodLength (think 120 is still a very sane value in modern wide screens. )
- Disable the warnings of ApplicationRecord and the warnings of update_column/attribute
- Namespace IndentHash and DotPosition under layout

Let me know your thoughts or other rules that I missed.

### CODE REVIEW

**Comment by kxmbrian** [comment]
Perhaps increase block length (especially for tests) and enforce use of `raise` instead of `fail`?

**Comment by allenwq** [comment]
@kxmbrian Updated the block length and raise/fail.

Block length is disabled because it does not make sense to limit the block length for our specs. and for normal application code, method/class length guards this case already.

File: app/controllers/course/announcements_controller.rb [hunk 1]
```diff
@@ -10,6 +10,8 @@ def index #:nodoc:
   end
 
   def show #:nodoc:
+    raise 'lol'
+    fail 'lol'
```
Comment (allenwq) [inline app/controllers/course/announcements_controller.rb]: 🎉

**Comment by kxmbrian** [comment]
When I run on the latest version of rubocop locally, I get quite a few of these: [SymbolArray](http://www.rubydoc.info/github/bbatsov/RuboCop/RuboCop/Cop/Style/SymbolArray) and [FilePath](http://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Rails/FilePath).

For FilePath, I think it makes sense to change our code.
For SymbolArray, perhaps we can put a MinSize of 4?

**Comment by allenwq** [comment]
@kxmbrian I further disabled `SymbolArray`, because our `WordArray` is also disabled and they should be consistent: https://github.com/Coursemology/coursemology2/pull/2371/files#diff-11a0d7906801d9dea0eccb85667ad811R62

**Pull request merged**